### PR TITLE
Implement Navigation Split View

### PR DESCRIPTION
**Pull request by rahulon12**
- Added support for iPadOS and macOS with a navigation Split View. I'd recommend running the branch locally and testing out different views in addition to looking through the diff.
- Refactor some existing code.

Simulator Screenshot - iPad Pro 11-inch (M4) - 2024-10-25 at 02 14 41
Screenshot 2024-10-25 at 2 15 08 AM

### CODE REVIEW

File: CanvasPlusPlayground/Views/AsyncAttributedText.swift [hunk 1]
```diff
@@ -9,18 +9,18 @@ import SwiftUI
...
     let htmlText: String
-    @State var announcementText:NSAttributedString? = nil
-    
+    @State var announcementText: NSAttributedString? = nil
+
     var body: some View {
         Group {
             if let announcementText {
-                AttributedText(announcementText)
+                Text(AttributedString(announcementText))
```
Comment (rahulon12) [inline CanvasPlusPlayground/Views/AsyncAttributedText.swift:17]: @iWatger got rid of the `AttributedString` view representable you were using and now using the native SwiftUI one. It appears slightly different but still functionally the same.

**Comment by azooz2003-bit** [comment]
On an unrelated note, the People tab still causes the app to freeze. We found the cause last time, still an issue though.

File: CanvasPlusPlayground/Helpers/PDFViewHelper.swift [hunk 2]
```diff
@@ -8,21 +8,41 @@
...
+    func makeView(context: Context) -> PDFView {
         let pdfView = PDFView()
-        async {
+        Task {
             pdfView.document = PDFDocument(url: self.pdfURL)
         }
         pdfView.autoScales = true
         return pdfView
     }
-
```
Comment (azooz2003-bit) [inline CanvasPlusPlayground/Helpers/PDFViewHelper.swift:22]: Application crashes upon loading a PDF.
CleanShot 2024-10-25 at 15 59 13@2x

Comment (azooz2003-bit) [reply CanvasPlusPlayground/Helpers/PDFViewHelper.swift:22]: Is the `Task` necessary here? Seems like it's attaching the document to the PDF Viewer on the BG thread after it's rendered on the main thread (via the SwiftUI view), hence the crash.

EDIT: Just tried it with `DispatchQueue.main.async` and it works fine. But the UI gets unresponsive for a small duration.

Comment (azooz2003-bit) [reply CanvasPlusPlayground/Helpers/PDFViewHelper.swift:22]: Ok, this works without issue.
```swift
Task {
    let document = PDFDocument(url: self.pdfURL)
    DispatchQueue.main.async {
        pdfView.document = document
    }
}
```

**Comment by azooz2003-bit** [comment]
Is this intentional? Can't change my access token after setting it.
CleanShot 2024-10-25 at 16 18 01

**Review by azooz2003-bit — CHANGES_REQUESTED** [review]
All LGTM! Just that PDF issue.

**Comment by rahulon12** [comment]
> On an unrelated note, the People tab still causes the app to freeze. We found the cause last time, still an issue though.

Fixed in #37

File: CanvasPlusPlayground/Helpers/PDFViewHelper.swift (hunk 2 above)
Comment (rahulon12) [reply CanvasPlusPlayground/Helpers/PDFViewHelper.swift]: Fixed

**Comment by rahulon12** [comment]
> Is this intentional? Can't change my access token after setting it. CleanShot 2024-10-25 at 16 18 01 [ CleanShot 2024-10-25 at 16 18 01 ](https://private-user-images.githubusercontent.com/67667005/380313555-36732bed-0018-4095-83a3-71143a090cd7.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4ODg2NTIsIm5iZiI6MTcyOTg4ODM1MiwicGF0aCI6Ii82NzY2NzAwNS8zODAzMTM1NTUtMzY3MzJiZWQtMDAxOC00MDk1LTgzYTMtNzExNDNhMDkwY2Q3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI1VDIwMzIzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYxM2U3YWQwZDhmNjU3YmQ2ZTM0NWM0ODcwZTBkYWZjYmE0NTI4NzJiOTMyYWU3ZmE1NjZiMmE2YzFmYjRjYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.kHFpU0T2MiCjgQIxSBHy4q9qcU08QI7bZi1m6LdkxSQ) [ ](https://private-user-images.githubusercontent.com/67667005/380313555-36732bed-0018-4095-83a3-71143a090cd7.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4ODg2NTIsIm5iZiI6MTcyOTg4ODM1MiwicGF0aCI6Ii82NzY2NzAwNS8zODAzMTM1NTUtMzY3MzJiZWQtMDAxOC00MDk1LTgzYTMtNzExNDNhMDkwY2Q3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI1VDIwMzIzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYxM2U3YWQwZDhmNjU3YmQ2ZTM0NWM0ODcwZTBkYWZjYmE0NTI4NzJiOTMyYWU3ZmE1NjZiMmE2YzFmYjRjYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.kHFpU0T2MiCjgQIxSBHy4q9qcU08QI7bZi1m6LdkxSQ)

I'm not running into this, can you create an issue?

**Review by azooz2003-bit — APPROVED** [review]

**Comment by iWatger** [comment]
This is really cool!

**Pull request merged**